### PR TITLE
Tweak transforms tutorial.

### PR DESCRIPTION
**Pull request by anntzer**
- In the Axes coordinates example, point to the "labelling subplots" example which is arguably a better approach for labelling (it ensures that the labels stay aligned even if the axes have different widths). The current example in the tutorial is still useful for pedagogical purposes, though.
- Shorten the entries in the transformations tables to avoid having overly long lines.

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

File: galleries/users_explain/artists/transforms_tutorial.py [hunk 1]
```diff
@@ -23,55 +23,51 @@
...
-|                |units depends on the back end. For |                                                   |
-|                |example it is pixels for Agg and   |                                                   |
-|                |points for svg/pdf.                |                                                   |
-+----------------+-----------------------------------+---------------------------------------------------+
-
-
-
-
++----------------+-----------------------------------+-----------------------------+
+|Coordinate      |Description                        |Transformation object        |
```
[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py:27]: That seems worse to me ("top right" and "bottom left" would be repeating each other even more obviously than now).

File: galleries/users_explain/artists/transforms_tutorial.py [hunk 2]
```diff
@@ -247,14 +243,15 @@
...
-# top right.  You can also refer to points outside the range, so (-0.1,
-# 1.1) is to the left and above your axes.  This coordinate system is
-# extremely useful when placing text in your axes, because you often
-# want a text bubble in a fixed, location, e.g., the upper left of the axes
-# pane, and have that location remain fixed when you pan or zoom.  Here
-# is a simple example that creates four panels and labels them 'A', 'B',
-# 'C', 'D' as you often see in journals.
+# your axes or subplot, (0.5, 0.5) is the center, and (1.0, 1.0) is the top
+# right.  You can also refer to points outside the range, so (-0.1, 1.1)
+# is to the left and above your axes.  This coordinate system is extremely
```
[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py:248]: Feel free to change it in a separate commit, but that not the point of this PR.

File: galleries/users_explain/artists/transforms_tutorial.py [hunk 3]
```diff
@@ -247,14 +243,15 @@
...
+# useful when placing text in your axes, because you often want a text bubble
+# in a fixed, location, e.g., the upper left of the axes pane, and have that
+# location remain fixed when you pan or zoom.  Here is a simple example that
+# creates four panels and labels them 'A', 'B', 'C', 'D' as you often see in
+# journals.  (A more sophisticated approach for such labeling is presented at
+# :doc:`/gallery/text_labels_and_annotations/label_subplots`.)
```
[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py:254]: I do want to keep them, as the point here is really to explain how transAxes work and not the "more sophisticated approach".

[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py:254]: > ... you do that

I do *what*?  Sorry, I'm really not following you here.

File: galleries/users_explain/artists/transforms_tutorial.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py:248]: go for it

File: galleries/users_explain/artists/transforms_tutorial.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (anntzer) [reply galleries/users_explain/artists/transforms_tutorial.py]: I don't agree, but also don't really care enough, so sure, I changed it.

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
I'll merge this; follow-ups inspired by reading these changes are welcome.

**Pull request merged**